pkg/rhelv2/ovalutil/rpm.go

```diff
@@ -223,3 +231,31 @@ func GetDefinitionType(def oval.Definition) (string, error) {
 	}
 	return match[1], nil
 }
+
+// ParseUnpatchedCVEComponents parse components and resolution states of these components in an OVAL definition
+func ParseUnpatchedCVEComponents(def oval.Definition) map[string]string {
+	resolutions := def.Advisory.Affected.Resolutions
+	if len(resolutions) == 0 {
+		return nil
+	}
+	result := make(map[string]string)
+
+	for _, resolution := range resolutions {
+		state := resolution.State
+		components := resolution.Components
+
+		for j := 0; j < len(components); j++ {
+			component := components[j]
```

Comment:
If the index is only used to get the component, you can use `for _, component := range` syntax.